spec/app/ics-027-interchain-accounts/README.md

```diff
@@ -20,110 +20,126 @@ On Ethereum, there are two types of accounts: externally owned accounts, control
+
+
+The `tryRunTx` method in the`InterchainAccountModule` interface defines the way to create an outgoing packet for a specific chain type. The chain type determines how the IBC account transaction should be constructed and serialised for the recieving chain. The sending side should know in advance how the recieving side expects the incoming IBC packet to be structured. 
+
+#### Recieving Interfacce
+
+`createAccount` defines the way to determine the account's address by using the port & channel id. A newly created interchain account must not conflict with an existing account. Therefore, the host chain (the chain that the account will live on) must keep track of which blockchains have created an interchain account in order to verify the transaction signing authority in `authenticateTx`. `createAccount` should be called in the `chanOpenTry` callback as part of the channel creation handshake process defined in [ics-4](https://github.com/cosmos/ibc/tree/master/spec/core/ics-004-channel-and-packet-semantics).
 
-interface IBCAccountPacketData {
-  type: Type
-  data: Uint8Array
+`authenticateTx` validates the transaction and checks that the signers in the transaction have the right permissions (are the owners of the account in question).`
+
+`runTx` executes a transaction after the transaction has been successfully authenticated.
+
+### Packet Data
+`InterchainAccountPacketData` contains an array of messages that an interchain account can run and a memo string that is sent to the recieving chain. The example below is defined as a proto encoded message but each chain can encode this differently. This should be clearly defined in the documentation for each chain specific implementation.
+
+```typescript
+message InterchainAccountPacketData  {
+    repeated google.protobuf.Any messages = 1;
+    string memo = 2;
 }
 ```
 
-The acknowledgement data type describes the type of packet data and chain id and whether the result code of processing and the data of result if it is needed, and the reason for failure (if any).
+The acknowledgement packet structure is defined as in [ics4](https://github.com/cosmos/cosmos-sdk/blob/v0.42.4/proto/ibc/core/channel/v1/channel.proto#L134-L147). The acknowledgement result should contain the chain-id and the address of the targetted interchain account. If an error occurs on the recieving chain the acknowledgement should contain the error message.
 
 ```typescript
-interface IBCAccountPacketAcknowledgement {
-    type: Type
-    code: uint32
-    data: Uint8Array
-    error: string
+message Acknowledgement {
+  // response contains either a result or an error and must be non-empty
+  oneof response {
+    bytes  result = 21;
```

Comment:
What do we want to send back to the sending chain on a successful `RunTx`? I specified above that we should send the chain-id and the address of the interchain account but I'm wondering what is actually necessary here.

Reply:
When we defined the [subcall standard in CosmWasm](https://github.com/CosmWasm/cosmwasm/blob/main/packages/std/src/results/subcall.rs#L52-L65) (to allow getting results from a call to another contract/module), we pass back the data field and the events returned.

A similar format is used in the tx responsed (data and events for each message).

Maybe events are too variable for returning over IBC, but I would at least include the data field (which is part of the ResultHash)

Reply:
I'm not entirely sure what an `event` would be in this context. Would you mind giving me an example?

Reply:
Every message in the sdk returns optional binary data and a list of events.

Events each have a type string, and a list of key-value attributes (also strings).

Eg.bank send emits "transfer" type with recipient, sender, and amount as attributes